analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,254 @@
+         Gene1B=gsub("[(].*","",standardFusioncalls$Gene1B),
+         Gene2B=gsub("[(].*","",standardFusioncalls$Gene2B))
+
+
+
+#############artifactFiltering#############
+
+
+fusion_filtering_QC<-function(standardFusioncalls=standardFusioncalls,readingFrameFilter=readingFrameFilter,artifactFilter=artifactFilter,junctionReadCountFilter=junctionReadCountFilter,spanningFragCountFilter=spanningFragCountFilter,reathroughFilter=reathroughFilter){
+  # @param standardFusioncalls A dataframe from star fusion or arriba standardized to run through the filtering steps
+  # @param expressionMatrix A rds with expression data (RSEM/TPM counts) for the same cohort/GTEX
+  # @param readingFramFilter A regex to capture readingframe (eg. inframe|frameshift|other)
+  # @param artifactFilter A red flag filter from Annotation ; in OpenPBTA annotation is from FusionAnnotator column: "annots"
+  # @param junctionReadCountFilter An integer threshold for JunctionReadCount
+  # @param spanningFragCountFilter An integer threshold for (SpanningFragCount - JunctionReadCount)
+  # @param expressionFilter An integer threshold for TPM/FPKM filter
+  # @return Standardized fusion calls filtered to pass QC and remove calls with insufficient read-support and annotation red-flags
+  
+  if( reathroughFilter & any(grepl('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE))){
+    # Gather read throughs from standardized fusion calls  
+    rts <- standardFusioncalls[grep('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE),"FusionName"]
+    # Reverse of read throughs to capture 
+    rts.rev <- unique(unlist(lapply(strsplit(rts, '--'), FUN = function(x) paste0(x[2],'--',x[1]))))
+    # Combine read through and reverse fusion genes 
+    rts <- unique(c(rts, rts.rev))
+    standardFusioncalls <- standardFusioncalls[-which(standardFusioncalls$FusionName %in% rts),]
+  }
+
+  if( !missing(readingFrameFilter ) ){
+    #TODO check for values
+    standardFusioncalls <- standardFusioncalls[grep(readingFrameFilter,standardFusioncalls$Fusion_Type),]
+  }
+  
+  if( !missing(artifactFilter) & any(grepl(artifactFilter,standardFusioncalls$annots)) ) {
+    #TODO check for values
+    standardFusioncalls <- standardFusioncalls[-grep(artifactFilter,standardFusioncalls$annots),]
+  }
+
+  if( !missing(junctionReadCountFilter ) ){
+    standardFusioncalls <- standardFusioncalls[which(standardFusioncalls$JunctionReadCount>=junctionReadCountFilter),]
+  }
+
+  if( !missing(spanningFragCountFilter ) ){
+    # false positive calls either at the start or end of a transcript will be supported by an uneven majority of support from spanning fragments compared to junction reads
+    # to remove these calls we are implementing this condition below
+    standardFusioncalls <- standardFusioncalls[which( (standardFusioncalls$SpanningFragCount - standardFusioncalls$JunctionReadCount ) <= spanningFragCountFilter),]
+  }
+  
+  return(standardFusioncalls)
+}
```

Comment:
If you're looking to move more to the `tidyverse` throughout:
```r
if( reathroughFilter & any(grepl('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE))){

  # Gather read throughs from standardized fusion calls
  rts <- standardFusioncalls %>%
    dplyr::filter(grepl('read.*through|NEIGHBORS', annots ,ignore.case = TRUE)) %>%
    dplyr::pull(FusionName)

  # Reverse of read throughs to capture
  rts.rev <-  unique(paste0(stringr::word(rts, 2, sep = "--"), "--",
                            stringr::word(rts, 1, sep = "--")))

  # Combine read through and reverse fusion genes
  rts <- unique(c(rts, rts.rev))

  # remove read through and reverse fusion genes
  standardFusioncalls <- standardFusioncalls %>%
    dplyr::filter(!(FusionName %in% rts))
}


if( !missing(readingFrameFilter ) ){
  #TODO check for values
  standardFusioncalls <- standardFusioncalls %>%
    dplyr::filter(grepl(readingFrameFilter ,Fusion_Type))
}

if( !missing(artifactFilter) & any(grepl(artifactFilter,standardFusioncalls$annots)) ) {
  #TODO check for values
  standardFusioncalls <- standardFusioncalls %>%
    dplyr::filter(!grepl(artifactFilter, annots))
}

if( !missing(junctionReadCountFilter ) ){
  standardFusioncalls <- standardFusioncalls %>%
    dplyr::filter(JunctionReadCount >= junctionReadCountFilter)
}

if( !missing(spanningFragCountFilter ) ){
  # false positive calls either at the start or end of a transcript will be supported by an uneven majority of support from spanning fragments compared to junction reads
  # to remove these calls we are implementing this condition below
  standardFusioncalls <- standardFusioncalls %>%
    dplyr::filter((SpanningFragCount - JunctionReadCount) <= spanningFragCountFilter)
}
```

I haven't tested these all though!